Deeply nested phrasing content is parsed incorrectly

rniwa opened the pull request:
#### 67d7bd2e0aa2ed0355e51c406ff56021e603590f
<pre>
Deeply nested phrasing content is parsed incorrectly
<a href="https://bugs.webkit.org/show_bug.cgi?id=162527">https://bugs.webkit.org/show_bug.cgi?id=162527</a>

Reviewed by Brent Fulgham.

Fixed the bug by implementing the missing early return in the adoption agency algorithm.

Based on <a href="https://crrev.com/69ad2d7d5eecfb79a8b5dc4d01b575926500ef08">https://crrev.com/69ad2d7d5eecfb79a8b5dc4d01b575926500ef08</a>

* LayoutTests/imported/w3c/web-platform-tests/html/syntax/parsing/adoption_agency_check_the_end_tag_name-expected.txt:
* Source/WebCore/html/parser/HTMLTreeBuilder.cpp:
(WebCore::HTMLTreeBuilder::callTheAdoptionAgency):

Canonical link: <a href="https://commits.webkit.org/252979@main">https://commits.webkit.org/252979@main</a>
</pre>

brentfulgham commented on Source/WebCore/html/parser/HTMLTreeBuilder.cpp:
```diff
@@ -1417,14 +1417,23 @@ void HTMLTreeBuilder::processAnyOtherEndTagForInBody(AtomHTMLToken&& token)
+        return;
+    }
+
+    // 4 is covered by the for() loop.
```
Are 1, 3, and 16 not handled by the for() loop?

rniwa replied:
Nope. The algorithm got refactored.

rniwa committed:
Deeply nested phrasing content is parsed incorrectly
https://bugs.webkit.org/show_bug.cgi?id=162527

Reviewed by Brent Fulgham.

Fixed the bug by implementing the missing early return in the adoption agency algorithm.

Based on https://crrev.com/69ad2d7d5eecfb79a8b5dc4d01b575926500ef08

* LayoutTests/imported/w3c/web-platform-tests/html/syntax/parsing/adoption_agency_check_the_end_tag_name-expected.txt:
* Source/WebCore/html/parser/HTMLTreeBuilder.cpp:
(WebCore::HTMLTreeBuilder::callTheAdoptionAgency):

Canonical link: https://commits.webkit.org/252979@main
